### PR TITLE
Sheerlike cleanup: deprecate global_include tag

### PR DESCRIPTION
**Pull request by chosak**
This change deprecates use of the sheerlike `global_include` tag that allowed for inclusion of arbitrary Jinja2 templates in Django templates. It replaces that tag with two specific tags that allow rendering of the cf.gov site header and footer in Django templates. The header and footer are themselves Jinja2 templates and so can't be included in Django templates without custom code.

In `v1.templatetags.header_footer` there are now two new Django template tags, `{% include_header %}` and `{% include_footer %}`, that render the site header and footer in a Django template (after `{% load header_footer %}` has been included in the template).

This change is intended to simplify the logic to include these modules. I've moved the templates from their previous `cfgov/jinja2/v1/_includes/on-demand` location to a more specific `v1/jinja2/v1` folder. This is more inline with Django conventions; as the Jinja2 engine is configured in settings with `'APP_DIRS': True`, the `jinja2` subdirectory is the conventional place for app templates to live.

Relatedly, I've removed the other remaining on-demand template, `secondary-navigation.html`, as I couldn't find any code that currently uses it:

https://github.com/search?q=user%3Acfpb+secondary-navigation.html&type=Code

(@anselmbradford @virginiacc does this seem right to you?)

To test locally, run a local server and visit http://localhost:8000/know-before-you-owe/. This is a Django template that includes the site header and footer, rendered using the Jinja2 template engine and these new tags.

See also new unit tests that validate behavior of these new template tags.

This change is part of a larger effort to remove the deprecated sheerlike app entirely from cfgov-refresh.

## Checklist

- [X] PR has an informative and human-readable title
- [X] Changes are limited to a single goal (no scope creep)
- [X] Code can be automatically merged (no conflicts)
- [X] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [X] Passes all existing automated tests
- [X] Any _change_ in functionality is tested
- [X] New functions are documented (with a description, list of inputs, and expected output)
- [X] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [X] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by hillaryj** [comment]
Back-end Travis testing failed with:

```
./cfgov/sheerlike/__init__.py:47:40: F821 undefined name 'get_request'
ERROR: InvocationError: '/home/travis/build/cfpb/cfgov-refresh/.tox/lint-py27/bin/flake8'
```

**Comment by chosak** [comment]
Thanks @hillaryj, fixed and rebased in 40771e3.

File: cfgov/jinja2/v1/_includes/on-demand/secondary-navigation.html [hunk 1]
```diff
@@ -1,9 +0,0 @@
-{# ==========================================================================
-
-   Render secondary navigation, for use in the global_include used
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/on-demand/secondary-navigation.html]: I think this removal is fine. For reference, the secondary nav is this on e.g. https://www.consumerfinance.gov/about-us/careers/current-openings/

screen shot 2018-07-09 at 3 14 10 pm

File: cfgov/v1/jinja2/v1/header.html [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {# ==========================================================================
 
-   Render a header without the banner, for use in the global_include used
-   in non-V1 projects.
+   Render the site header without the banner.
```
[the remark this answers is not in the document]
Comment (anselmbradford) [reply cfgov/v1/jinja2/v1/header.html]: Maybe add some info here of how and why this is used and differs from that used in base.html

**Pull request merged**